pkgs/top-level/go-packages.nix

```diff
@@ -2671,13 +2665,21 @@ let
       pushover
     ];
 
-    buildFlagsArray = ''
+    # Tests exist, but seem to clash with the firewall.
+    doCheck = false;
+
+    preBuild = ''
+      export GO15VENDOREXPERIMENT=1
```

Comment:
We're building binaries, not libraries, and they're statically linked. My own inclination is that the Prometheus devs know what they're doing, so let's use the version of the dependencies that they developed and tested with.